src/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java

```diff
@@ -149,6 +149,16 @@ public void setValue(ReactSlider view, double value) {
     view.setOnSeekBarChangeListener(null);
     view.setValue(value);
     view.setOnSeekBarChangeListener(ON_CHANGE_LISTENER);
+    if (view.isAccessibilityFocused() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+      int index = (int)value;
+      String sliderValue = view.mAccessibilityIncrements.get(index);
+      int stringLength = view.mAccessibilityUnits.length();
+      String spokenUnits = view.mAccessibilityUnits;
+      if (sliderValue != null && Integer.parseInt(sliderValue) == 1) {
+        spokenUnits = spokenUnits.substring(0, stringLength - 1);
+      }
+      view.announceForAccessibility(String.format("%s %s", sliderValue, spokenUnits));
```

Comment:
As a follow up to comment above this is how it would look like if utilized the `setupAccessibility`.
```suggestion
      int index = (int)value;
      view.setupAccessibility(index);
```

Of course, the `setupAccessibility` should be refactored so that it takes an argument of index to use, and the call of this method should be then modified to use `(int)mValue` when called in `ReactSlider` class or `(int)value` when called here.

Reply:
modified as per suggestion, thank you!